### PR TITLE
feat: use frontend-plugin-framework to provide a FooterSlot

### PR DESCRIPTION
**Pull request by brian-smith-tcril**
Note: This removes the ability to use `LOGO_POWERED_BY_OPEN_EDX_URL_SVG` to set the logo in the footer. The footer component itself supports using `LOGO_TRADEMARK_URL` to set the logo.

Note: There are other `PluginSlots` in this MFE that have not been moved to the `plugin-slots` directory and turned into components yet. I wanted to keep the focus of this PR on adding the `FooterSlot`, so I created an issue to track that effort https://github.com/openedx/frontend-app-learner-dashboard/issues/344 and will PR those changes separately

### CODE REVIEW

**Comment by jsnwesson** [comment]
Thanks for writing this up, Braden! The @openedx/2u-aperture team and I will be looking over this PR early next week, while also monitoring [the conversation for Learning MFE's PR](https://github.com/openedx/frontend-app-learning/pull/1381).

File: src/App.jsx [hunk 1]
```diff
@@ -109,7 +109,7 @@ export const App = () => {
               )}
           </main>
         </AppWrapper>
-        <Footer logo={getConfig().LOGO_POWERED_BY_OPEN_EDX_URL_SVG} />
+        <FooterSlot />
```
Comment (arbrandes) [inline src/App.jsx:112]: Can you please also remove the var from https://github.com/openedx/frontend-app-learner-dashboard/blob/master/.env.development#L11? and https://github.com/openedx/frontend-app-learner-dashboard/blob/master/.env.test#L11?

**Review by arbrandes — APPROVED** [review]
Looks good, though we should remove the env var from all the files before merging.

File: src/App.jsx (hunk 1 above)
Comment (brian-smith-tcril) [reply src/App.jsx:112]: Done!

**Review by jsnwesson — APPROVED** [review]

**Pull request merged**